packages/fxa-settings/src/pages/ResetPasswordRedesign/CompleteResetPassword/container.tsx

```diff
@@ -0,0 +1,195 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+
+import { useValidatedQueryParams } from '../../../lib/hooks/useValidate';
+import {
+  Integration,
+  useAccount,
+  useConfig,
+  useFtlMsgResolver,
+} from '../../../models';
+import { KeyStretchExperiment } from '../../../models/experiments';
+
+import CompleteResetPassword from '.';
+import {
+  AccountResetData,
+  CompleteResetPasswordLocationState,
+} from './interfaces';
+import GleanMetrics from '../../../lib/glean';
+import firefox from '../../../lib/channels/firefox';
+import { getLocalizedErrorMessage } from '../../../lib/auth-errors/auth-errors';
+import { useState } from 'react';
+
+// This component is used for both /complete_reset_password and /account_recovery_reset_password routes
+// for easier maintenance
+
+const CompleteResetPasswordContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const keyStretchExperiment = useValidatedQueryParams(KeyStretchExperiment);
+
+  const account = useAccount();
+  const config = useConfig();
+  const ftlMsgResolver = useFtlMsgResolver();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const searchParams = location.search;
+
+  if (!location.state) {
+    navigate(`/reset_password${searchParams}`);
+  }
+
+  const {
+    code,
+    email,
+    token,
+    accountResetToken,
+    emailToHashWith,
+    kB,
+    recoveryKeyId,
+  } = location.state as CompleteResetPasswordLocationState;
+
+  const hasConfirmedRecoveryKey = !!(
+    accountResetToken &&
+    email &&
+    kB &&
+    recoveryKeyId
+  );
+
+  const isResetWithoutRecoveryKey = !!(code && token);
+
+  const handleNavigationWithRecoveryKey = () => {
+    navigate(`/reset_password_with_recovery_key_verified${location.search}`);
```

Comment:
Just curious, in other places you reference `searchParams` (which come from `location.search`), but here you reference `location.search`. Is there a reason for not using searchParams here?

Reply:
No reason, just missed the update.